kratos/python_scripts/testing/run_tests.py

```diff
@@ -181,108 +161,53 @@ def print_summary(exit_codes):
-                sys.exit()
-
-            for a in parsedApps:
-                if a not in applications + ['KratosCore']:
-                    print('Warning: Application {} does not exist'.format(a))
-                    sys.exit()
-
-            applications = parsedApps
-
-            if 'KratosCore' in applications:
-                applications.remove('KratosCore')
-
-        elif o in ('-c', '--command'):
-            try:
-                cmd = str(a)
-            except:
-                print('Error: Cannot parse command name {0}.'.format(a))
-                Usage()
-                sys.exit()
-
-        elif o in ('--using-mpi'):
-            is_mpi = True
-
-        elif o in ('-t', '--timer'):
-            try:
-                timer = int(a)
-            except:
-                print('Error: Cannot parse command name {0}.'.format(a))
-                Usage()
-                sys.exit()
+    # parse command line options
+    parser = argparse.ArgumentParser()
 
-        else:
-            assert False, 'unhandled option'
+    parser.add_argument('-c', '--command', default=cmd, help="Use the provided command to launch test cases. If not provided, the default \'runkratos\' executable is used")
+    parser.add_argument('-l', '--level', default='all', choices=['all', 'nightly', 'small', 'validation'], help="Minimum level of detail of the tests: \'all\'(Default) \'(nightly)\' \'(small)\'")
+    parser.add_argument('-v', '--verbosity', default=1, type=int, choices=[0, 1, 2], help="Verbosity level: 0, 1 (Default), 2")
+    parser.add_argument('-a', '--applications', default=applications, choices=applications, help="List of applications to run separated by \':\'. All compiled applications will be run by default")
+    parser.add_argument('-m', '--using-mpi', default=False, help="If running in MPI and executing the MPI-tests")
+    parser.add_argument('-t', '--timer', default=-1, help="Use the provided custom time limit for the execution. If not provided, the default values are used")
 
-    if is_mpi:
+    args = parser.parse_args()
+
+    # Set if mpi
+    if args.using_mpi:
         level = "mpi_" + level
 
+    # Parser the applications
+    if args.applications is str:
+        parsedApps = args.applications.split(':')
+    else:
+        parsedApps = args.applications
+    for a in parsedApps:
+        if a not in applications + ['KratosCore']:
+            print('Warning: Application {} does not exist'.format(a))
+            sys.exit()
+    applications = parsedApps
+    if 'KratosCore' in applications:
+        applications.remove('KratosCore')
+
     # Set timeout of the different levels
     signalTime = None
-    if timer > 0:
-        signalTime = timer
+    if int(args.timer) > 0:
+        signalTime = int(args.timer)
     else:
-        if level == 'small':
-            signalTime = int(90)
-        elif level == 'nightly':
-            signalTime = int(900)
+        if args.level == 'small':
+            signalTime = 90

```

Comment:
I would prefer to keep the cast here if possible- It gives problem with the signals in win (see https://github.com/KratosMultiphysics/Kratos/commit/dbf821e7b468cf85d2150abf92a014c857752f0d)

Reply:
Okay

Reply:
Done